shared/api/storage_pool_volume.go

```diff
@@ -60,6 +60,12 @@ type StorageVolumePost struct {
 	//
 	// API extension: storage_api_remote_volume_snapshots
 	VolumeOnly bool `json:"volume_only" yaml:"volume_only"`
+
+	// Source project name
+	// Example: foo
+	//
+	// API extension: storage_volume_project_move
+	SourceProject string `json:"source_project,omitempty" yaml:"source_project,omitempty"`
```

Comment:
I'm confused why is this `SourceProject`?

The API call is to the volume in the current project, so it feels like this should be `Project` (same as with instances) and be used to pass what project we want it moved to.

Reply:
@stgraber,

when we move storage volume the API call is to the volume in destination project because of this:

https://github.com/lxc/lxd/blob/c44d34710798c231f28a9622ebd58a48a668d451/lxc/storage_volume.go#L390-L392

so we need pass source project. Actually it is very similar to what we have in instance copying:

We set project for destServer:

https://github.com/lxc/lxd/blob/c44d34710798c231f28a9622ebd58a48a668d451/lxc/copy.go#L113-L115

and then in `CopyInstance` we do:
https://github.com/lxc/lxd/blob/c44d34710798c231f28a9622ebd58a48a668d451/client/lxd_instances.go#L453


Instance moving is a little bit different because we don't call `UseProject` for the destination server, so in API project argument point to the source project and we pass targetProject in InstancePost.

So you are right that there is some inconsistency between moving storage volumes and instances.

Reply:
ok, so maybe we should do this:
https://github.com/lxc/lxd/blob/fa40c1700df9785c9973d33e1e152ebdfeb9e0bc/lxc/storage_volume.go#L390-L392
only when we copy storage volume (then copying storage volumes and instance will be consistent).

and when we move instance just pass target project to `StorageVolumePost` without calling `dstServer.UseProject` on client side. Sounds reasonable?

Reply:
The `UseProject` trick should only be needed during copy or during move between two servers. For those, the move is effectively a copy (which requires a POST to create the instance or volume in the target project) and then once copied, delete the source.

This issue is about the optimized server-side move, for those, we should really only need to POST to the existing instance or volume and pass `Project: "some-value"` in the POST data, then LXD will handle it server-side. As that request is done against the existing instance or volume, it's done against the source project and the target project is a field in the POST data.


So the short answer is that yes, we should do the same as instances and only have to use UseProject if dealing with a copy, not with move.